app/Exceptions/ViewRedirect.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exceptions;
+
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+/**
+ * @deprecated TODO do not redirect in views, refactor to controller when needed
```

Comment:
To clarify just for my own understanding & intellectual curiosity, is it okay for controllers to force some kind of redirect to occur?

Reply:
Absolutely, yes. if anything it should _only_ be middlewares and controllers that return redirect responses as they are in charge of handling Http requests.
views on the other hand expect the request to be processed except for rendering the response.

That said, Laravel comes with a lot of mechanisms that cause redirects automatically. E.g. when a middleware checks for authentication but a user accessed it as a guest it will remember the current page, redirect to the login page, and redirect back to the intended URL after successful login.